### ISSUE TITLE
Saved Neural Network loses information.

### ISSUE THREAD

**Issue by jacobdwatters** [jacobdwatters/JML#42]
- Dropout Layers: When a neural network is saved, dropout layers are not saved in the file. 
- Optimizers: When a neural network is saved, the optimizer used during training is not saved.

This means when the model is reloaded, it will be missing these components of the neural network. Neither of these will result in different predictions being made by the loaded model but if the model was to be retained, it may not behave as the user expects.

To address this, the optimizer's details (including learning rate scheduling) and dropout layers should be saved to the .mdl file so they can be read upon loading the model and the model can be constructed correctly.
